src/ext/dispatch_compat_php5.c

```diff
@@ -227,4 +227,57 @@ void ddtrace_forward_call(zend_execute_data *execute_data, zval *return_value TS
     zend_fcall_info_args_clear(&fci, 1);
     zval_dtor(&args);
 }
+
+static zend_function *_get_current_fbc(zend_execute_data *execute_data TSRMLS_DC) {
+    if (EX(opline)->opcode == ZEND_DO_FCALL_BY_NAME) {
+        return FBC();
+    }
+    zend_op *opline = EX(opline);
+    zend_function *fbc = NULL;
+    zval *fname = opline->op1.zv;
+
+    if (CACHED_PTR(opline->op1.literal->cache_slot)) {
+        return CACHED_PTR(opline->op1.literal->cache_slot);
+    } else if (EXPECTED(zend_hash_quick_find(EG(function_table), Z_STRVAL_P(fname), Z_STRLEN_P(fname) + 1,
+                                             Z_HASH_P(fname), (void **)&fbc) == SUCCESS)) {
+        return fbc;
+    } else {
+        return NULL;
+    }
+}
+
+int ddtrace_should_trace_call(zend_execute_data *execute_data, zend_function **fbc,
```

Comment:
Nice, much cleaner than the previous approach!
One suggestion - some time ago BOOL_T define/type was introduced and started to be used in new code. If you feel this approach has merits feel free to use BOOL_T and FALSE, TRUE defines. for functions expected to return true and false values only.

Reply:
Thanks @pawelchcki! I like the idea of using `BOOL_T` - done! :)